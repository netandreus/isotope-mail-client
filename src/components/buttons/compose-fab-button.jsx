import React from 'react';
import PropTypes from 'prop-types';
import {translate} from 'react-i18next';
import mainCss from '../../styles/main.scss';

export const ComposeFabButton = ({t, onClick}) => (
  <button className={`${mainCss['mdc-fab']} ${mainCss['compose-fab-button']}`} onClick={onClick}>
    <span className={`material-icons ${mainCss['mdc-fab__icon']}`}>edit</span>
    <span className={`${mainCss['compose-fab-button__label']}`}>{t('composeButton.compose')}</span>
  </button>
);

ComposeFabButton.propTypes = {
  onClick: PropTypes.func
};

ComposeFabButton.defaultProps = {
  onClick: () => {}
};

export default translate()(ComposeFabButton);
