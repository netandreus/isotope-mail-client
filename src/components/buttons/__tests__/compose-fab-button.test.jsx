import React from 'react';
import {shallow} from 'enzyme/build/index';
import {ComposeFabButton} from '../compose-fab-button';


describe('ComposeFabButton component test suite', () => {
  test('Snapshot render, should render compose-fab-button', () => {
    // Given
    const props = {t: key => key, onClick: jest.fn()};
    // When
    const composeFabButton = shallow(<ComposeFabButton {...props} />);
    // Then
    expect(composeFabButton).toMatchSnapshot();
  });
  test('click, should trigger function', () => {
    // Given
    const props = {t: key => key, onClick: jest.fn()};
    const composeFabButton = shallow(<ComposeFabButton {...props} />);
    // When
    composeFabButton.find('button').simulate('click');
    // Then
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
