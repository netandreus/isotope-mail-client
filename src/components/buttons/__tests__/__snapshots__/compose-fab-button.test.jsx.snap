// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComposeFabButton component test suite Snapshot render, should render compose-fab-button 1`] = `
<button
  className="mdc-fab compose-fab-button"
  onClick={[MockFunction]}
>
  <span
    className="material-icons mdc-fab__icon"
  >
    edit
  </span>
  <span
    className="compose-fab-button__label"
  >
    composeButton.compose
  </span>
</button>
`;
